src/test/resources/com/puppycrawl/tools/checkstyle/checks/imports/unusedimports/InputUnusedImportsFromStaticMethodRef.java

```diff
@@ -0,0 +1,94 @@
+/*
+UnusedImports
+processJavadoc = (default)true
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.imports.unusedimports;
+
+import static java.lang.annotation.ElementType.TYPE_USE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.Objects;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Collections;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import static java.lang.String.format; // violation 'Unused import - java.lang.String.format.'
+import static java.util.Arrays.sort; // violation 'Unused import - java.util.Arrays.sort.'
+import static java.util.List.of; // violation 'Unused import - java.util.List.of.'
+import static java.util.Collections.emptyMap; // violation 'Unused import - java.util.Collections.emptyMap.'
+
+/**
+ * Use {@link Collections::emptyMap} in this javadoc.
+ */
+public class InputUnusedImportsFromStaticMethodRef {
+
+    public static <T, R> Function<T, R> func(Function<T, R> f) { return f; }
+
+    InputUnusedImportsFromStaticMethodRef() {
+    }
+
+    void testMethodRef()
+    {
+        Optional<String> test = Optional.empty();
+        test.map(String::format);
```

Comment:
can you make one more Input where same code is used but without "import" and it compiles and shows no violation.
So we have a prove that user can simply remove unused import.

Reply:
Added a new test `testMethodRefNotImported` to `InputUnusedImportsFromStaticMethodRef.java`

Reply:
Thanks a lot, this is also valuable, keep it.
But thought about "without static imports", but with regular import.